balancer-js/src/modules/pools/pool-types/concerns/composableStable/exit.concern.ts

```diff
@@ -16,20 +83,68 @@ export class ComposableStablePoolExit implements ExitConcern {
     wrappedNativeAsset,
     singleTokenMaxOut,
   }: ExitExactBPTInParameters): ExitExactBPTInAttributes => {
-    // TODO implementation
-    console.log(
-      exiter,
-      pool,
+    this.checkInputsExactBptIn(
       bptIn,
-      slippage,
-      shouldUnwrapNativeAsset,
-      wrappedNativeAsset,
-      singleTokenMaxOut
+      singleTokenMaxOut,
+      pool,
+      shouldUnwrapNativeAsset
     );
 
-    throw new Error('To be implemented');
-  };
+    const sortedValues = this.getValuesSorted({
+      pool,
+      wrappedNativeAsset,
+      shouldUnwrapNativeAsset,
+      singleTokenMaxOut,
+    });
+
+    const { parsedTokens } = sortedValues;
+    // Parse pool info into EVM amounts in order to match amountsIn scalling
+    const expectedAmountsOut: string[] = Array(parsedTokens.length).fill('0');
+    const minAmountsOut = Array(parsedTokens.length).fill('0');
+    let userData: string;
 
+    if (
+      singleTokenMaxOut &&
+      !isUndefined(sortedValues.singleTokenMaxOutIndex)
+    ) {
+      this.doMathsV1ExactBPTInSingleTokenOut(
+        sortedValues,
+        expectedAmountsOut,
```

Comment:
We don't need to use these in the maths function.
Make the maths return the token out amount (minus slippage) then create `expectedAmountsOut` and `minAmountsOut` after.